pkg/provider/azure_loadbalancer.go

```diff
@@ -1296,6 +1296,73 @@ func getIdleTimeout(s *v1.Service) (*int32, error) {
 	return &to32, nil
 }
 
+func getProbeIntervalInSecondsAndNumOfProbe(s *v1.Service) (*int32, *int32, error) {
+	// get number of probes
+	numberOfProbes, err := getInt32FromAnnotations(s.Annotations, consts.ServiceAnnotationLoadBalancerHealthProbeNumOfProbe, func(val *int32) error {
+		//total probe should be less than 120, minimum probe interval is 2 and minimum number of unhealthy responses is 2
+		//number of probe values should range between 2 to 60
+		if *val > 60 || *val < 2 {
```

Comment:
Could you get the interval first and validate the total probe interval by `number*interval`?

Reply:
Moved this business rule out of validator function and check it at the end of the function, which decouples the validator functions.